lib/alternatives.js

```diff
@@ -81,7 +81,7 @@ internals.Alternatives.prototype.when = function (ref, options) {
     var obj = this.clone();
     var is = Cast.schema(options.is);
 
-    if (options.is === null || !options.is.isJoi) {
```

Comment:
This change doesn't look correct as the logic has fundamentally changed with your changes.

I think instead you meant to change this to:
```js
if (options.is === null || options.is === undefined || !options.is.isJoi)
```

That way you fall into it when you get passed an `undefined`, `null`, or something that isn't already Joi.